sdk/dotnet/Pulumi/Deployment/Deployment_Invoke.cs

```diff
@@ -55,7 +55,22 @@ private async Task<OutputData<T>> RawInvoke<T>(string token, InvokeArgs args, In
             // the provider fully responsible for depdendency
             // tracking, which is a good future direction also for
             // `Invoke`.
-            var result = await InvokeRawAsync(token, args, options).ConfigureAwait(false);
+
+            var keepResources = await this.MonitorSupportsResourceReferences().ConfigureAwait(false);
+            var serializedArgs = await SerializeInvokeArgs(token, args, keepResources);
+
+            // Short-circuit actually invoking if `Unknowns` are
+            // present in `args`, otherwise preview can break.
+            if (serializedArgs.PropertyValues.Values.Any(Serializer.ContainsUnknowns))
```

Comment:
Nit: Wondering if it could be:

```suggestion
            if (Serializer.ContainsUnknowns(serializedArgs.PropertyValues))
```

Shorter and avoids the enumerator allocation for `Values` and the delegate allocation for the method group passed to `Any`.

Reply:
IT can since PropertyValues happens to be ImmutableDictionary<string,object> which is allowed by the grammar understood by ContainsUnknowns. I was typechecking it using a different type in my head though, have not thought of it.